synapse/storage/databases/main/event_push_actions.py

```diff
@@ -580,7 +591,7 @@ def _get_receipts_by_room_txn(
 
         args.extend((user_id,))
         txn.execute(sql, args)
-        return cast(List[Tuple[str, int]], txn.fetchall())
+        return dict(cast(List[Tuple[str, int]], txn.fetchall()))
```

Comment:
nit: Separating this a bit will help with readability.

```suggestion
        return {
            room_id: latest_stream_ordering
            for room_id, latest_stream_ordering in txn.fetchall()
        }
```